src/server/common/server_base.cpp

```diff
@@ -129,20 +129,20 @@ void server_base::serve(boost::asio::yield_context yield, boost::asio::ip::tcp::
 		return;
 	}
 
-#ifndef _WIN32
-	if(keep_alive_) {
-		int timeout = 30;
+	int timeout = 300;
 #ifdef __linux__
-		int cnt = 10;
-		int interval = 30;
-		setsockopt(socket->native_handle(), SOL_TCP, TCP_KEEPIDLE, &timeout, sizeof(timeout));
-		setsockopt(socket->native_handle(), SOL_TCP, TCP_KEEPCNT, &cnt, sizeof(cnt));
-		setsockopt(socket->native_handle(), SOL_TCP, TCP_KEEPINTVL, &interval, sizeof(interval));
-#endif
-#if defined(__APPLE__) && defined(__MACH__)
-		setsockopt(socket->native_handle(), IPPROTO_TCP, TCP_KEEPALIVE, &timeout, sizeof(timeout));
-#endif
-	}
+	int cnt = 10;
+	int interval = 30;
+	setsockopt(socket->native_handle(), SOL_TCP, TCP_KEEPIDLE, &timeout, sizeof(timeout));
+	setsockopt(socket->native_handle(), SOL_TCP, TCP_KEEPCNT, &cnt, sizeof(cnt));
+	setsockopt(socket->native_handle(), SOL_TCP, TCP_KEEPINTVL, &interval, sizeof(interval));
+#elif defined(__APPLE__) && defined(__MACH__)
+	setsockopt(socket->native_handle(), IPPROTO_TCP, TCP_KEEPALIVE, &timeout, sizeof(timeout));
+#elif defined(_WIN32)
+	// these are in milliseconds for windows
+	timeout *= 1000;
+	setsockopt(socket->native_handle(), SOL_SOCKET, SO_RCVTIMEO, (const char*)&timeout, sizeof(timeout));
+	setsockopt(socket->native_handle(), SOL_SOCKET, SO_SNDTIMEO, (const char*)&timeout, sizeof(timeout));
```

Comment:
Does this work? https://learn.microsoft.com/en-us/windows/win32/api/winsock/nf-winsock-setsockopt says the type should be a `DWORD`. Is that guaranteed to be compatible to `int`? It's probably rather an unsigned type.

I would prefer `reinterpret_cast` instead of a C-cast.

Reply:
The documentation at the top says
```
int setsockopt(
  [in] SOCKET     s,
  [in] int        level,
  [in] int        optname,
  [in] const char *optval,
  [in] int        optlen
);
```
so I'm not sure where a DWORD comes into play?

I will change it to be reinterpret_cast though.

Reply:
If you scroll further down you can see a table with `optname` to type mappings. `SO_RCVTIMEO` and `SO_SNDTIMEO` are listed as `DWORD` there.

Reply:
That documentation seems rather inconsistent then, but that would mean instead of
```
timeout *= 1000;
setsockopt(socket->native_handle(), SOL_SOCKET, SO_RCVTIMEO, reinterpret_cast<const char*>&timeout, sizeof(timeout));
setsockopt(socket->native_handle(), SOL_SOCKET, SO_SNDTIMEO, reinterpret_cast<const char*>&timeout, sizeof(timeout));
```
it'd be
```
DWORD timeout_ms = timeout * 1000;
setsockopt(socket->native_handle(), SOL_SOCKET, SO_RCVTIMEO, &timeout_ms, sizeof(timeout_ms));
setsockopt(socket->native_handle(), SOL_SOCKET, SO_SNDTIMEO, &timeout_ms, sizeof(timeout_ms));
```
?

Reply:
Yeah, I would say that is safer.